jerry-core/vm/vm.c

```diff
@@ -136,8 +144,20 @@ vm_op_set_value (ecma_value_t object, /**< base object */
 
     if (ECMA_IS_VALUE_ERROR (to_object))
     {
+      ecma_free_value (to_object);
+
+#ifdef JERRY_ENABLE_ERROR_MESSAGES
+      ecma_value_t error_value = ecma_raise_standard_error_with_format (ECMA_ERROR_TYPE,
+                                                                        "Cannot set property '%' of %",
+                                                                        property,
+                                                                        object);
+#else /* !JERRY_ENABLE_ERROR_MESSAGES */
+      ecma_value_t error_value = ecma_raise_type_error (NULL);
+#endif /* JERRY_ENABLE_ERROR_MESSAGES */
+
       ecma_free_value (property);
-      return to_object;
+
+      return error_value;
```

Comment:
I see the point of recreating the error when messages are enabled, but not when they are disabled. Please update this ifdef to not change the original behavior.